packages/rest/README.md

```diff
@@ -52,16 +52,37 @@ Application options.
 const app = new RestApplication({
   rest: {
     port: 3001,
+
+    // Template of the OpenAPI spec, which will be filled with `paths` from
+    // controllers.
+    openApiSpec: {
```

Comment:
I believe we already have an API for setting the initial OpenAPI spec values, see http://apidocs.loopback.io/@loopback%2fdocs/rest.html#RestApplication.prototype.api

Can we keep using `app.api()` instead of introducing this new option? If not, then what is your reasoning? What benefits do you see in this declarative approach?

The current downside of `app.api()` is that the user has to provide `spec.openapi` and `spec.paths` properties in addition to `info`.

```ts
app.api({
  openapi: '3.0.1',
  info: { /*...*/ },
  servers: [/*...*/],
  paths: {},
});
```

On the brighter side, `app.api()` allows TypeScript to verify the correctness of the passed values. I am not sure if the type checks work for your declarative approach, because the type used for options allow arbitrary (additional) properties.

Reply:
We can relax the constraints on the argument of `app.api()` to allow any subset of OpenAPISpec and fill in `openapi` version and an empty `paths` object ourselves.

Reply:
See also https://github.com/strongloop/loopback-next/pull/1637#issuecomment-417295302

Reply:
Doesn't `app.api()` take in an entire spec vs. the new `openApiSpec` property which is intended to accept a base template for the spec rather than the complete spec itself (which would come from `app.api`)?

Reply:
I believe our current code merges the following sources:
 - the spec provided by `app.api()`  (if any)
 - specs provided at controller level via `@api` decorator,
 - specs built from controller-method-level decorators like `@get` and `@param`
 - specs provided via `app.route(new Route(verb, path, spec, ...))`

Reply:
I don't mind providing a declarative way for providing a base template for the OpenAPI spec, perhaps in addition to `app.api()` API. However, I think we need to move that declarative config to a different place than application/server config, see https://github.com/strongloop/loopback-next/pull/1637#issuecomment-417295302.